src/plugins/intel_gna/src/gna_plugin.cpp

```diff
@@ -665,145 +667,169 @@ void GNAPlugin::AddDebugProperties(const InferenceEngine::CNNLayerPtr layer,
+    // Operations Max and Min aren't supported
+    pass_config->disable<ngraph::pass::ConcatReduceFusion>();
+    // pass_config->disable<ngraph::pass::SoftSignDecomposition>();
+    manager.run_passes(model);
+}
+
 void GNAPlugin::LoadNetwork(const CNNNetwork& _network) {
     OV_ITT_SCOPED_TASK(itt::domains::GNAPlugin, "LoadNetwork");
     std::shared_ptr<InferenceEngine::details::CNNNetworkImpl> convertedNetwork;
 
     std::string effectiveGnaCompileTargetValue = effectiveGnaCompileTarget();
 
-    bool isNgraphPassesUsed = false;
+    bool is_ngraph_passes_used = false;
     bool fake_quantized = false;
 
     if (_network.getFunction()) {
         CNNNetwork clonedNetwork = InferenceEngine::cloneNetwork(_network);
-        const auto& graph = clonedNetwork.getFunction();
-        ngraph::pass::Manager manager;
-        manager.register_pass<ngraph::pass::InitNodeInfo>();
-
-        fake_quantized = ngraph::op::util::has_op_with_type<ngraph::opset7::FakeQuantize>(graph);
-        // In OV API 2.0(IRv10) default convertion to fp32 (inputs, outputs and weights) is disabled
-        // and we need to run the ConvertPrecision transformation to support old networks.
-        manager.register_pass<ngraph::pass::ConvertPrecision>(
-            precisions_array{{ngraph::element::f16, ngraph::element::f32}});
-        manager.register_pass<ngraph::pass::ConvertMVN1ToMVN6>();
-        manager.register_pass<ov::intel_gna::pass::DecomposeMVN>();
-        manager.register_pass<ngraph::pass::CommonOptimizations>();
-        manager.register_pass<ov::intel_gna::pass::RemoveInputConvert>();
-        manager.register_pass<ov::intel_gna::pass::RemoveOutputConvert>();
-        manager.register_pass<ngraph::pass::ConvertSequenceToTensorIterator>();
-        manager.register_pass<ngraph::pass::GRUCellDecomposition>();
-        manager.register_pass<ngraph::pass::LSTMCellDecomposition>();
-        manager.register_pass<ov::intel_gna::pass::ConvertDWSCToScaleShifts>();
-        manager.register_pass<ov::intel_gna::pass::ConvertPaddedToValidConv>();
-        manager.register_pass<ov::intel_gna::pass::Decompose2DConvTransposedWithBiasAF>(effectiveGnaCompileTargetValue, config.gnaPrecision);
-        manager.register_pass<ov::intel_gna::pass::Decompose2DConvTransposedWithBias>(effectiveGnaCompileTargetValue, config.gnaPrecision);
-        manager.register_pass<ov::intel_gna::pass::Decompose2DConv>(effectiveGnaCompileTargetValue, config.gnaPrecision);
-        // TODO enable this transformation for networks with convolutions
-        if (!ngraph::op::util::has_op_with_type<ngraph::opset7::Convolution>(graph)) {
-            manager.register_pass<ov::intel_gna::pass::ConvertMatmulWithFqToPointWiseConvolution>();
-            manager.register_pass<ov::intel_gna::pass::ConvertMatmulWithBiasToPointWiseConvolution>();
-            manager.register_pass<ov::intel_gna::pass::ConvertMatmulToPointWiseConvolution>();
+        const auto& model = clonedNetwork.getFunction();
```

Comment:
I would change it to
``` 
auto model = clonedNetwork.getFunction();
```
due the fact method does not return reference to internal element but new shared_ptr.

Reply:
Added check for ptr